lib/jobs/ipmi-job.js

```diff
@@ -312,42 +228,164 @@ function ipmiJobFactory(
+                            var mask = 1 << 7; //jshint ignore: line
+                            for (var i = hexBuffer.length-1; i >= 0 ; i = i-16) {
+                                //Reading the buffer from end to start. The loop exits when all
+                                ///the needed info is processed, it avoids processing information
+                                // that is not needed
+                                index1 = i-3;
+                                index2 = i-5;
+                                if (index1 < hexBuffer.length && counter !== 0) {
+                                    //This is from table 32 of the IPMI spec v4
+                                    var eventType = (hexBuffer[index1] & ~mask).toString(16).slice(-2);// jshint ignore:line
+                                    eventType = pad.substring(0, pad.length - eventType.length) + eventType;// jshint ignore:line
+                                    var sensorType = hexBuffer[index2].toString(16);
+                                    sensorType = pad.substring(0, pad.length - sensorType.length) + sensorType;// jshint ignore:line
+                                    decodedSel["Event Type Code"] = eventType;
+                                    decodedSel["Sensor Type Code"] = sensorType;
+
+                                    newEntries[counter-1]= _.merge(newEntries[counter-1], decodedSel);// jshint ignore:line
+                                }
+                                else{
+                                    //Done getting the information needed for the new sel entries
+                                    //No need to process the rest of the buffer
+                                    break;//jshint ignore: line
+                                }
+                                counter = counter -1;
+                            }
+
+                            workObject.config.lastPreviouslyReadSelEntry = allEntries[allEntries.length - 1]["SEL Record ID"];// jshint ignore:line
+                            workObject.config.lastSelDeleteTimeLastRun = lastSelDeleteTimeCurrent;
+                            workObject.save();
+                            return workObject.save();
+                        })
+                        .then(function(){
+                            logger.debug("***number of new SEL entries :" + newEntries.length +
+                                ". Poller: "+data.workItemId+ " nodeID: " + workObject.node );//jshint ignore: line
+                            return newEntries; //returning the new  SEL entries
+                        }).
+                        catch(function(err){
+                            logger.error("Failed to process SEL. Poller: " +data.workItemId +
+                                " nodeID: " + workObject.node + ". Error message: " + err.message);
+                        })
+                        .finally(function () {
+                            if(filePath){
```

Comment:
Truth sir! However, it introduces a cluster of other cleanup paths to deal with. Really, we should use 'tmp' rather than 'temp', but this is critical path. :)